### ISSUE TITLE
JSON Logs Do Not Contain References to klog.kObj

### ISSUE THREAD

**Issue by seanmalloy** [kubernetes-sigs/descheduler#426]
This bug was introduced in pull request #412 as part of the descheduler v0.20.0 release cycle.

Testing the logging-format flag was not quite straight forward. To keep a long story short:

I couldn't figure out how to pass the flag to `kube-scheduler`. Using the binary by itself requires authentication/some complicated bootstrap process.

```
bin$ ./kube-scheduler --logging-format=json
I1013 00:14:58.458921  353074 serving.go:331] Generated self-signed cert in-memory
W1013 00:14:58.705507  353074 authentication.go:265] No authentication-kubeconfig provided in order to lookup client-ca-file in configmap/extension-apiserver-authentication in kube-system, so client certificate authentication won't work.
```

I went with the `kubelet` as I can pass the flag when creating a cluster with kind, with the following config:

```
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
kubeadmConfigPatches:
- |
  apiVersion: kubelet.config.k8s.io/v1beta1
  kind: KubeletConfiguration
  logging:
    format: json
nodes:
- role: control-plane
- role: worker
- role: worker
- role: worker
```

Problem is structured logging is not used in `pkg/kubelet` so I replaced [this](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_node_status.go#L92) line with:

```
klog.Infof("random message")
klog.ErrorS(err, "Unable to register node with API server", "node", klog.KObj(node))
```

and built an image and loaded it with kind. ([tutorial](https://mauilion.dev/posts/kind-k8s-testing/))

[Here](https://gist.githubusercontent.com/farah/34486f980418a99c08cca8cb6909768d/raw/d854cea895d2f0d82f7deffc027a22692b24a687/kubelet%2520logs) are the logs.

To summarise, given this log:
`klog.ErrorS(err, "Unable to register node with API server", "node", klog.KObj(node))`

The json output is:

`{"ts":1602507889919.717,"msg":"Unable to register node with API server","v":0,"err":"Post \"https://master-control-plane:6443/api/v1/nodes\": dial tcp 172.18.0.2:6443: connect: connection refused"}`

So yes it seems anything that uses objects in the log is not expected to be output.

Which is quite strange.

_Originally posted by @farah in https://github.com/kubernetes-sigs/descheduler/pull/412#issuecomment-707120912_

**Comment by seanmalloy** [comment]
Related to #347

**Comment by seanmalloy** [comment]
This appears to be a bug with the upstream `k8s.io/klog/v2` module and not the a bug with the descheduler itself. Specifically the https://pkg.go.dev/k8s.io/klog/v2#KObj function.

**Comment by damemi** [comment]
@seanmalloy I don't think the comment you have linked now is relevant for us --- `connection refused` errors are usually more infra/deployment related. So I wouldn't refer to that comment's text. I think this is more relevant: https://github.com/kubernetes-sigs/descheduler/pull/412#issuecomment-707158341 because it shows the error: `klog.KObj` doesn't print in JSON logging.

I'm also more suspicious that this should be opened as a bug upstream in k/k, rather than here. If it's an issue with the `KObj` function than it should be opened against klog, not the descheduler
